alphagov/paas-uaa-customized-boshrelease
# Fix the Crown logo for our users

What
----

We're passing in the Crown logo on our pages with an inline SVG block.
This is great, as less resources are called in to render the page the
way we want it to be. SVGs have the fallback URL, just in case a browser
is not capable of supporting the embedded code.

This causes us two issues:

1. Microsoft Edge was attempting to load both resources (SVG and the
fallback PNG)
2. The PNG was returning a 302, which redirects the user to the login
page

The Design System team has addressed the issue number 1 and Ollie did an
amazing job of explaining the problem in the [commit message][1].

This is more of an accessibility and reliability issue.

The one we've been facing is number 2.

We're generating a CSRF token, on each load of the forms element for
security purposes. Each time the page is loaded, it's passing the token
into the form as well as setting up a cookie for the user, just to
validate that token elsewhere later.

If the page is being loaded twice, that means:
- a token is being passed in HTML and set as a cookie
- a token is being set as a cookie - reset, no longer matching HTML

Therefore after a form submission, the code that was sent via the form,
does not match the code that has been sent via the cookie.

This is weirdly solved, by providing a fallback URL which will in fact
resolve to a PNG or a 404, instead of more HTML.

[1]: https://github.com/alphagov/govuk-frontend/pull/1337/commits/84485f589523be4101fd1912b1e2e91af7f9a3bf

How to review
-------------

- Sanity check
- Make sure the image is accessible without auth
- See if that's the correct place to be fixing that
- See [rafalp](https://login.rafalp.dev.cloudpipeline.digital/login) in Microsoft Edge
